packages/mui-base/src/Portal/Portal.tsx

```diff
@@ -67,7 +67,7 @@ const Portal = React.forwardRef(function Portal(
       {mountNode ? ReactDOM.createPortal(children, mountNode) : mountNode}
     </React.Fragment>
   );
-});
+}) as React.ForwardRefExoticComponent<PortalProps & React.RefAttributes<Element>>;
```

Comment:
Should MUI X adopt this pattern too? ([ForwardRefExoticComponent source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8ef783a008b722934e43c18669b33cf3b4e3be43/types/react/index.d.ts#L776))

Reply:
Yes, it should help, apparently.
I figured out that the offending part is the `PropsWithoutRef<P>` type that's used in the `forwardRef` return type signature. It's responsible for the `Pick<>` [you were wondering about](https://github.com/mui/material-ui/issues/35287#issuecomment-1408870399).

cc @mui/x

Reply:
Right now we are doing the following on X:

```tsx
type DesktopDatePickerComponent = (<TDate>(
  props: DesktopDatePickerProps<TDate> & React.RefAttributes<HTMLDivElement>,
) => JSX.Element) & { propTypes?: any };

const DesktopDatePicker = React.forwardRef(...) as DesktopDatePickerComponent
```

The React type seems to be slightly more complete but the idea remains the same.